DesyncSafeAnalyzer/DesyncSafeAnalyzer/Shared.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using DesyncSafeAnalyzer.Tools;
+using Microsoft.CodeAnalysis;
+
+namespace DesyncSafeAnalyzer
+{
+  public static class Shared
+  {
+    private static readonly List<string> SafeWc3Natives = new List<string>
+    {
+      "BlzSetSpecialEffectColor",
+      "BlzSetSpecialEffectColorByPlayer",
+      "StartSound",
+      "PingMinimap",
+      "SetCameraField",
+      "GetRectMinX",
+      "GetRectMinY",
+      "GetRectMaxX",
+      "GetRectMaxY",
+      "SetCameraBounds",
+      "BlzChangeMinimapTerrainTex",
+      "CameraSetupApplyForceDuration",
+      "SetMinimapIconVisible",
+      "GetUnitX",
+      "GetUnitY",
+      "GetItemX",
+      "GetItemY",
+      "ElementAt",
+      "GetRandomInt",
+      "CreateMinimapIcon",
+      "SkinManagerGetLocalPath",
+      "QuestSetEnabled",
+      "ClearTextMessages",
+      "GetPlayerId",
+      "FourCC",
+      "BlzFrameSetVisible"
+    };
+
+    private static readonly List<string> SafeDotNetNatives = new List<string>
```

Comment:
This only cares about the function name, so a custom declared function with this name will always be considered desync-safe. That seems incredibly risky; many of these function names are common and could be accidentally used.